library/kani/kani_lib.c

```diff
@@ -2,11 +2,82 @@
 // SPDX-License-Identifier: Apache-2.0 OR MIT
 
 #include <assert.h>
+#include <stdbool.h>
 #include <stddef.h>
 #include <stdint.h>
 #include <stdlib.h>
 #include <string.h>
 
+// `assert` then `assume`
+#define __KANI_assert(cond, msg)     \
+    do {                             \
+        bool temp = (cond);          \
+        __CPROVER_assert(temp, msg); \
+        __CPROVER_assume(temp);      \
+    } while (0)
```

Comment:
Can you just use brackets here to create a new scope instead of `do {} while (0)`?

Reply:
`do {} while (0)` is the standard way of ensuring that a `;` is both required and works in the expected way.